packages/babel-traverse/src/path/conversion.js

```diff
@@ -223,10 +223,20 @@ function hoistFunctionEnvironment(
     );
 
     argumentsPaths.forEach(argumentsChild => {
-      const argsRef = t.identifier(argumentsBinding);
-      argsRef.loc = argumentsChild.node.loc;
-
-      argumentsChild.replaceWith(argsRef);
+      let currentScope = argumentsChild.scope;
+      do {
+        if (
+          !currentScope.hasOwnBinding("arguments") &&
+          !(
+            currentScope.path.isFunctionExpression() ||
+            currentScope.path.isFunctionDeclaration()
```

Comment:
```suggestion
            currentScope.path.isFunction()
```
A class method is also a function.
```js
var twelve = () => {
  var arguments = 2;
  return class {
    m() { return () => arguments; }
  }
};
```

Reply:
👍🏻

Reply:
Ah the ArrowFunctionExpression should be excluded.
```js
currentScope.path.isFunction() && !currentScope.path.isArrowFunctionExpression()
```